manager/integration/tests/test_settings.py

```diff
@@ -777,14 +781,171 @@ def test_setting_concurrent_rebuild_limit():  # NOQA
+
+    # Step 1-9
+    update_setting(client,
+                   "concurrent-replica-rebuild-per-node-limit",
+                   "1")
+
+    # Step 1-10
+    delete_replica_on_test_node(client, volume1_name)
+    wait_for_rebuild_start(client, volume1_name)
+    volume1 = client.by_id_volume(volume1_name)
+    current_node = get_self_host_id()
+    replicas = []
+    for r in volume1.replicas:
+        if r["hostId"] == current_node:
+            replicas.append(r)
+
+    assert len(replicas) > 0
+    crash_replica_processes(client, core_api, volume1_name, replicas)
+    delete_replica_on_test_node(client, volume2_name)
+
+    # While volume 2 is rebuilding, verify that volume 1 is not
+    # rebuilding and stuck in degrading state
+    wait_for_rebuild_start(client, volume2_name)
+    for i in range(RETRY_COUNTS):
+        volume1 = client.by_id_volume(volume1_name)
+        volume2 = client.by_id_volume(volume2_name)
+
+        if volume2.rebuildStatus == []:
+            break
+
+        assert volume2.rebuildStatus[0].state == "in_progress"
+        assert volume1.rebuildStatus == []
+
+        time.sleep(RETRY_INTERVAL)
+
+    wait_for_rebuild_complete(client, volume2_name)
+    wait_for_rebuild_start(client, volume1_name)
+    wait_for_rebuild_complete(client, volume1_name)
+
+    # Step 2-1
+    # Step 2-2
+    wait_for_volume_healthy(client, volume1_name)
+    wait_for_volume_healthy(client, volume2_name)
+
+    volume2 = client.by_id_volume(volume2_name)
+    lht_host_id = get_self_host_id()
+    volume2.detach(hostId=lht_host_id)
+
+    # Step 2-2
+    delete_replica_on_test_node(client, volume1_name)
+    wait_for_rebuild_start(client, volume1_name)
+
+    # Step 2-3
+    volume2 = client.by_id_volume(volume2_name)
+    volume2.attach(hostId=lht_host_id)
+
+    # In a 2 minutes retry loop:
+    # verify that we can see the case: volume2 becomes healthy while
+    # volume1 is rebuilding
+    expect_case = False
+    for i in range(RETRY_COUNTS):
+        volume1 = client.by_id_volume(volume1_name)
+        volume2 = client.by_id_volume(volume2_name)
+
+        try:
+            if volume1.rebuildStatus[0].state == "in_progress" and \
+                    volume2["robustness"] == "healthy":
+                expect_case = True
+                break
+        except: # NOQA
+            pass
+        time.sleep(RETRY_INTERVAL)
+    assert expect_case is True
+
+    wait_for_volume_healthy(client, volume1_name)
+    volume2.detach(hostId=lht_host_id)
+    wait_for_volume_detached(client, volume2_name)
+
+    volume2.attach(hostId=lht_host_id)
+    wait_for_volume_healthy(client, volume2_name)
+
+
+def delete_replica_on_test_node(client, volume_name): # NOQA
+
+    lht_host_id = get_self_host_id()
+
+    volume = client.by_id_volume(volume_name)
+    for replica in volume.replicas:
+        if replica.hostId == lht_host_id:
+            replica_name = replica.name
+    volume.replicaRemove(name=replica_name)
```

Comment:
This should be a reusable function. Please reorganize this when doing refactoring in the future.